pytorch_lightning/loops/fit_loop.py

```diff
@@ -149,19 +139,15 @@ def _results(self) -> _ResultCollection:
 
     @property
     def done(self) -> bool:
-        """Evaluates when to leave the loop.
-
-        Returns True if trainer.should_stop was set (e.g. by early stopping) or if the maximum number of steps or epochs
-        is reached.
-        """
+        """Evaluates when to leave the loop."""
         # TODO(@awaelchli): Move track steps inside training loop and move part of these condition inside training loop
         stop_steps = _is_max_limit_reached(self.global_step, self.max_steps)
-        stop_epochs = _is_max_limit_reached(self.current_epoch, self.max_epochs)
+        stop_epochs = _is_max_limit_reached(self.epoch_progress.current.completed, self.max_epochs)
```

Comment:
the drawback is that it becomes less readable.

Reply:
can be just `self.trainer.current_epoch` since it's just a property now?

Reply:
This is on purpose because the follow-up will use `processed`